lib/ansible/executor/task_executor.py

```diff
@@ -831,7 +839,7 @@ def _poll_async_result(self, result, templar, task_vars=None):
 
         if int(async_result.get('finished', 0)) != 1:
             if async_result.get('_ansible_parsed'):
-                return dict(failed=True, msg="async task did not complete within the requested time - %ss" % self._task.async_val)
+                return dict(failed=True, msg="async task did not complete within the requested time - %ss" % self._task.async_val, async_result=async_result)
```

Comment:
Note: added `async_result` here so that we can get access to the async job ID used in display output in the default callback plugin.